src/data.ts

```diff
@@ -0,0 +1,36 @@
+import * as Vue from 'vue'
+import { VueClass } from './declarations'
+
+const noop = () => {}
+
+export function collectDataFromConstructor (vm: Vue, Component: VueClass) {
+  // Create dummy Vue instance to collect
+  // initial class properties from the component constructor.
+  // To prevent to print warning,
+  // the data object should inherit Vue.prototype.
+  const data = Object.create(Component.prototype, {
+    _init: {
+      get: () => noop
+    }
+  })
+
+  // proxy to each prop values
+  const propKeys = Object.keys(vm.$options.props || {})
+  propKeys.forEach(key => {
```

Comment:
Thanks for your review! I'm thinking this problem but maybe we can solve this by using `vm` as the prototype for `data` instead of `Component.prototype`?